tests/kernel/mem_protect/userspace/testcase.yaml

```diff
@@ -3,8 +3,8 @@ tests:
     filter: CONFIG_ARCH_HAS_USERSPACE
     tags: kernel security userspace ignore_faults
     min_ram: 36
-  kernel.memory_protection.userspace.armv8m_gap_filling:
-    arch_whitelist: arm
+  kernel.memory_protection.userspace.gap_filling:
+    arch_whitelist: arm arc
```

Comment:
do we even need this whitelist? seems like CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS filter ought to be enough now

Reply:
already changed.
it makes sense to make it more generic to covery any  archs which support CONFIG_MPU_REQUIRES_NON_OVERLAPPING_REGIONS.